### ISSUE TITLE
Discripency between matlab SOM and TorchSOM?

### ISSUE THREAD

**Issue by Mr-Nobody-dey** [michelin/TorchSOM#7]
In the well-established MATLAB SOM model (https://github.com/ilarinieminen/SOM-Toolbox), batch SOM did not have a learning rate, but here it does. Did the learning rate used to train the batch SOM?

**Comment by LouisTier** [comment]
Hello @Mr-Nobody-dey ,

## Short answer

Yes, the learning rate `α` is used in every weight update, and that is intended.

The apparent contradiction comes from the word **"batch"** meaning two *different algorithms*:

- In **SOM-Toolbox**, "batch" is the **Kohonen batch-map**: a learning-rate-free, closed-form update where each prototype is replaced by a neighborhood-weighted **mean** of the data.
- In **TorchSOM**, "batch" is **mini-batch vectorization of the online (stochastic) rule**: the online update is computed for every sample in a mini-batch and applied as a single averaged step, so it *does* carry a learning rate.

They are two members of the same family, not the same algorithm. 

## The three update rules

With $\alpha(t)$ the learning rate at epoch $t$, and $B$ the mini-batch size.

**1. Online / Stochastic SOM**: one update per sample (this is the rule shown in the README):

$$\mathbf{w}_{ij} \leftarrow \mathbf{w}_{ij} + \alpha(t)\ \cdot h_{ij}^{(n)}\ \cdot \bigl(\mathbf{x}_n - \mathbf{w}_{ij}\bigr)$$

**2. Kohonen batch-map (SOM-Toolbox)**: one closed-form pass over all $N$ samples, **no** `α`:

$$\mathbf{w}_{ij} \leftarrow \frac{\sum_{n=1}^{N} h_{ij}^{(n)}\ \cdot \mathbf{x}_n}{\sum_{n=1}^{N} h_{ij}^{(n)}}$$

**3. TorchSOM mini-batch**: the online step summed over a mini-batch of $B$ samples, **with** `α`:

$$\mathbf{w}_{ij} \leftarrow \mathbf{w}_{ij} + \frac{\alpha(t)}{B} \sum_{n=1}^{B} h_{ij}^{(n)}(t)\ \cdot \bigl(\mathbf{x}_n - \mathbf{w}_{ij}\bigr)$$

|                     | Update granularity | Learning rate | Order-dependent |
| ------------------- | ------------------ | ------------- | --------------- |
| Online / stochastic | per sample         | yes           | yes             |
| Kohonen batch-map   | per full epoch     | **no**        | no              |
| **TorchSOM**        | per mini-batch     | **yes**       | no (within batch) |

## Why TorchSOM keeps α

Each mini-batch produces one aggregated update, which is the mean of the online steps over that
batch, so it needs `α` as its step size. The update lives in `_update_weights`
([`torchsom/core/som.py`](https://github.com/michelin/TorchSOM/blob/ba440acccb8ec7cf073cdf49df3f0212671426f7/torchsom/core/som.py#L241)):

```python
neighborhood_sum = neighborhoods.sum(dim=0)                      # Σ_n h
weighted_data = torch.einsum("bxy,bf->xyf", neighborhoods, data) # Σ_n h · x_n
updates = (weighted_data - neighborhood_sum.unsqueeze(-1) * self.weights) * (
    learning_rate / batch_size
)
self.weights.data += updates
```

Substituting the two sums recovers the mini-batch rule above:

$$\frac{\alpha}{B}\Bigl(\textstyle\sum_n h_{ij}^{(n)} \mathbf{x}_n - \bigl(\sum_n h_{ij}^{(n)}\bigr)\mathbf{w}_{ij}\Bigr) = \frac{\alpha}{B}\sum_{n} h_{ij}^{(n)}\bigl(\mathbf{x}_n - \mathbf{w}_{ij}\bigr)$$

Both `α` and the neighborhood width `σ` (used in $$h_{ij}^{(n)}(t)$$) decay across epochs.

**Comment by Mr-Nobody-dey** [comment]
Thanks for the reply, now I understand.
